active/ImageIcon/ImageIconSpec.md

```diff
@@ -0,0 +1,143 @@
+
+BitmapIcon supports a set of bitmap types such as png and jpeg
+(the same set supported by 
+[BitmapImage](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Media.Imaging.BitmapImage)).
+The general [Image](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Image)
+element on the other hand supports these and more types, plus supports loading types from
+a stream rather than just URI.
+
+The new API in this spec is an `ImageIcon` type, which like the other icons is an IconElement
+(so can be used where IconElement is required, such as 
+[NavigationViewItem.Icon](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.NavigationViewItem.Icon)).
+Other than the base class, ImageIcon is equivalent to Image.
+
+ImageIcon is a superset of the existing BitmapIcon minus the
+[BitmapIcon.ShowAsMonochrome](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.BitmapIcon.ShowAsMonochrome)
+property, as it is not a requirement from design to have all icons be monochrome.
+Since ShowAsMonochrome will not be part of ImageIcon, the 
+[IconElement.Foreground](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.IconElement.Foreground)
+inherited from IconElement will not be suported in this control.
+It will be up to the application to update the icon to match the user's theme. 
+
+Along with the various IconElement types, which are elements, there are IconSource types.
+For example BitmapIcon and
+[BitmapIconSource](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.BitmapIconSource).
+Similarly in this spec, along with `ImageIcon` is an `ImageIconSource`.
+
+In summary:
+
+```
+IconElement
+    BitmapIcon
+    ImageIcon **new
+
+IconSource
+    BitmapIconSource
+    ImageIconSource **new
+
+FrameworkElement
+    Image
+```
+
+# API Pages
+
+## ImageIcon class
+
+Represents an icon that uses an image as its content.
+
+ImageIcon can be used just like an
+[Image](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Image)
+element, except it derives from
+[IconElement](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.IconElement)
+and can therefore be used anywhere an IconElement is required, such as
+[NavigationViewItem.Icon](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.NavigationViewItem.Icon).
+
+> Note: the ImageIcon.Foreground property is currently ignored.
+
+## Is this the right control? 
+
+Use an ImageIcon control when you want to display an image as your icon. Do not use an ImageIcon if you want to display an icon from a font. 
+
+## Examples
+Below is an example of how you can use ImageIcon in an App Bar Button Item.
+
+> Spec note: This example will also be added to the App Bar Button Item page with the rest of the icon controls. 
+
+XAML
+```xml
+<AppBarButton Label="ImageIcon" Click="AppBarButton_Click">
+    <AppBarButton.Icon>
+        <ImageIcon Source="Wheel.png"/>
+    </AppBarButton.Icon>
+</AppBarButton>
+```
+
```

Comment:
I might not understand your question but the name came from following the naming convention we already have for the other WinUI icon controls where it is named based on the type of icon you are adding.

Reply:
I guess my problem is that I don't understand the mental model. We have images, icons, image sources, and icon sources. What's the difference? How do they fit together? (And sometimes the sources have other sources.)

Reply:
Let me give a stab at explaining it all:

Xaml has a concept of IconElements which are UIElements which display a small UI component. They don't have an interaction model, they aren't accessible etc. They are intended to be used in a composite scenario, like an AppBarButton.  Since UIElements can only have one parent, they are inherently  not sharable. So we have published IconSources; which are objects that contain all of the information needed to construct an IconElement.  These objects can be shared and anyone who wants can create their own IconElement from ann IconSource.

We have a bunch of different IconElement types (SymbolIcon, FontIcon, PathIcon, BitmapIcon, and now ImageIcon). And with each of those IconElement types we also publish IconSource types (SymbolIconSource, etc) which have the properties needed to create their IconElement counterpart.

Xaml Also has the concept of an image, which is a UIElement which displays some graphics.  Image has a source property of type ImageSource which is used to inform the Image where/how to generate the graphics it is to display.  There are a bunch of different image source types (SVGImageSource, BitmapImageSource, etc). These image sources instruct Image how to get the graphics to display.

The IconSource and ImageSource models are different; Image requires an image source which instructs the image where to get the graphics it is to display.  IconElements decide what to display based on their properties, for example FontIcon has: FontFamily, FontSize, FontStyle, FontWeight, and glyph properties, which it uses on an internal textbox to display the appropriate graphics. IconSource's is simply an object which contains the properties that its IconElement counterpart has, so that the object can be referenced to create new IconElements from.

The term Source is overloaded between these two controls, which is being exacerbated by the fact that the newest implementer of IconElement has Image in its name.